Fix presentation to focus on ALPN ID rather than SVCB records (#4)
anr-bmbf-pivot/draft-lenders-core-coap-dtls-svcb: draft-lenders-core-coap-dtls-svcb.md

informative:
  RFC8323: coap-tcp
  I-D.ietf-core-dns-over-coap: doc
  RFC4944: 6lo

--- abstract

This document specifies an Application-Layer Protocol Negotiation (ALPN) ID for
transport-layer-secured CoAP services.

--- middle

# Introduction

Application-Layer Protocol Negotiation (ALPN) allows agreeing on an application-layer protocol during a Transport Layer Security (TLS) handshake.
For that an ALPN ID is used.

> Furthermore, the ALPN ID can be used to discover Service Bindings (SVCB) via DNS, using SVCB resource records with the "alpn" Service Parameter Keys.

Reviewer: ```suggestion
Furthermore, the ALPN ID can discovered as part of Service Bindings (SVCB) via DNS, using SVCB resource records with the "alpn" Service Parameter Keys.
```

AIU they are only a result of the lookup, not part of the lookup. (The service identifier, eg. _dns, is used to discover the binding).

As an example, this information can be obtained as part of the discovery of DNS over CoAP (DoC) servers (see {{-doc}}) that deploy TLS or DTLS to secure their messages.

> This document specifies ALPN IDs for CoAP services that are secured by transport security, namely TLS and DTLS.

Reviewer: ```suggestion
This document specifies an ALPN ID for CoAP services that are secured by transport security, namely DTLS.
An APLPN ID for CoAP service secured by TLS has already been specified in {{?RFC8323}}.
```

# Terminology

> {::boilerplate bcp14-tagged}

Reviewer: We don't use the bcp14 identifiers at all in the document.

Author: Anything we should add to the Terminology section otherwise in your opinion? If not, I would just drop it completely.

Reviewer: I'm not aware of anything that warrants a terminology section.

# Application-Layer Protocol Negotiation (ALPN) IDs

For CoAP over TLS an ALPN ID was defined in {{-coap-tcp}}.
As it is not advisable to re-use the same ALPN ID for a different transport layer, an ALPN for
CoAP over DTLS is also registered in {{iana}}.

ALPN ID values have no fixed length.
A short value ("co") is allocated for CoAP over DTLS.

> This avoids fragmentation of Client Hello and Server Hello messages in constrained networks with link-layer fragmentation, such as 6LoWPAN {{-6lo}}.

Reviewer: ```suggestion
This can avoid fragmentation of Client Hello and Server Hello messages in constrained networks with link-layer fragmentation, such as 6LoWPAN {{-6lo}}.
```

We don't have concrete evidence of this being possible right now. (If our last experiments were optimized correctly, an ALPN wouldn't have made a difference; if cTLS hits, that should be different).
